### PR TITLE
Edit Haddocks, round 1

### PR DESCRIPTION
**Pull request by curiousleo**
Update function and typeclass Haddocks.

### CODE REVIEW

File: System/Random.hs [hunk 1]
```diff
@@ -312,13 +315,10 @@
 module System.Random
   (
 
-  -- $intro
```
Comment (curiousleo) [inline System/Random.hs:315]: There is no `$intro`.

File: System/Random.hs [hunk 2]
```diff
@@ -312,13 +315,10 @@
...
-
   -- * Random number generators
 
     RandomGen(..)
   , MonadRandom(..)
-  , Frozen(..)
```
Comment (curiousleo) [inline System/Random.hs]: `Frozen` is exported as part of `MonadRandom`.

File: System/Random.hs [hunk 3]
```diff
@@ -470,75 +486,144 @@ class RandomGen g where
...
   genRange _ = (minBound, maxBound)
 
-  -- | The 'split' operation allows one to obtain two distinct random number
-  -- generators.
+  -- | Returns two distinct pseudo-random number generators.
+  --
+  -- Implementations should take care to ensure that the resulting generators
+  -- are not correlated. Some pseudo-random number generators are not
+  -- splittable. In that case, the 'split' implementation should fail with a
+  -- descriptive 'error' message.
```
Comment (curiousleo) [inline System/Random.hs:530]: Do you agree with this guidance? Is there a more elegant way to deal with this?

**Comment by curiousleo** [comment]
Fixing conflicts now.

**Comment by curiousleo** [comment]
I'd like to do this in fairly quick iterations. Please review, I'll address the comments on this PR or a follow-up, there is still more documentation to be written / improved.

File: System/Random.hs (hunk 3 above)
Comment (lehins) [reply System/Random.hs:530]: I think the only elegant way to this problem is splitting RandomGen into two classes as discussed in #7 Until and if that will happen, the `error` is the best we can hope for.

**Review by lehins — APPROVED** [review]
Changes look great

File: System/Random.hs (hunk 3 above)
Comment (curiousleo) [reply System/Random.hs:530]: :+1:

**Pull request merged**